### ISSUE TITLE
Is using a connection Pool Recommended?

### ISSUE THREAD

**Issue by coderhs** [njh/ruby-mqtt#157]
Sorry this is not an issue but a question. If I am wrong to do it here my apologies, kindly point me in the right direction.

I am working with a ruby on rails application, that communicates using MQTT to manage devices. When ever we want to send a message we are pretty much using the below code.

```rb
    MQTT::Client.connect(host: ENV['MOSQUITTO_HOST'],
                         port: 1883,
                         username: ENV['MOSQUITTO_USERNAME'],
                         password: ENV['MOSQUITTO_PASSWORD']) do |c|
      c.publish(channel, msg)
    end
```

So they are connection and then pretty much finishing it, after its done. 

I recently started upgrading our app to Rails 6, and found that redis gem recommends using connection pool rather than having a single connection or multiple connection.

Is that recommended for MQTT connections as well? Has anyone done this? and if so how was your experience.

**Comment by cvalerio** [comment]
I totally agree a connection pool should be recommended. Or at least you should disconnect and dispose of your client, if you're done with it. One can't rely on garbage collector to dispose of the connection, because you cannot be sure of when the GC actually happen for the new client you just created (if it happens at all).

I recently came across an issue with the implementation of mqtt connection in one of the projects I follow as production manager: the mosquitto server was quickly depleting its available resouces, in terms of open connections. That was because the Rails application wasn't closing the connection when was done publishing, and it was creating a new one for each message that need to be published.
The result was that after a couple of days, mosquitto wasn't accepting any other connection, until I restarted the rails app container.
Lucky me, that happened in staging environment.

Ideally, since MQTT connections are long-lived (at least when you're listening), I expect a maximum of 4-5 open connection *per application*. Publish connection should be open and closed (if not used in long-running processes).
Using a pool would at least reuse opened connections.
